app/src/main/java/com/glia/exampleapp/ExampleCustomMessageAdapter.java

```diff
@@ -0,0 +1,87 @@
+package com.glia.exampleapp;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.Typeface;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.res.ResourcesCompat;
+
+import com.glia.androidsdk.chat.ChatMessage;
+import com.glia.widgets.UiTheme;
+import com.glia.widgets.chat.adapter.CustomMessageAdapter;
+import com.glia.widgets.chat.adapter.holder.CustomMessageViewHolder;
+import com.glia.widgets.chat.adapter.holder.WebViewViewHolder;
+
+import org.json.JSONException;
+
+public class ExampleCustomMessageAdapter extends CustomMessageAdapter {
```

Comment:
Based on this class. Do I understand correctly, that if the integrator wants to have a custom adapter then he would have to create custom views for all messages and there is no way to fall back to the default message rendering?

Reply:
No. Integrators will have the ability to create custom views for messages which contain the metadata. Other message types provide by SDK.

Reply:
Any way to fall back to default message rendering? If for example, this metadata does not affect the UI of the message and is for passing some internal information?

Reply:
Oh, got the answer for the javadocs 👍